pkg/apis/apps/fuzzer/fuzzer.go

```diff
@@ -40,6 +40,12 @@ var Funcs = func(codecs runtimeserializer.CodecFactory) []interface{} {
 			if len(s.Spec.UpdateStrategy.Type) == 0 {
 				s.Spec.UpdateStrategy.Type = apps.RollingUpdateStatefulSetStrategyType
 			}
+			if s.Spec.PersistentVolumeClaimPolicy == nil {
+				s.Spec.PersistentVolumeClaimPolicy = &apps.StatefulSetPersistentVolumeClaimPolicy{
+					OnSetDeletion: apps.RetainPersistentVolumeClaimDeletePolicyType,
```

Comment:
why are we matching the default instead of fuzzing? (random values)

Reply:
That's what the pod management policy & update strategy are doing. Not doing this meant tests broke.

Reply:
I looks like what the existing code is doing is incorrect. The fuzzer should give random values. See https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/batch/fuzzer/fuzzer.go for example.

Reply:
Done. Let me know if that's not what you had in mind.